### PR TITLE
Refactor preprocessor to incorporate into main program

### PR DESCRIPTION
**Pull request by ctessum**

### CODE REVIEW

**Comment by ctessum** [comment]
@dpaolella would you be willing to take a look at this and see if you can see any problems?

**Comment by dpaolella** [comment]
Sure, I can take a look!

File: inmap/configExampleGEOSChem.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+OutputFile= "${GOPATH}/src/github.com/spatialmodel/inmap/inmap/testdata/preproc/xxx.shp"
+
+EmissionUnits= "tons/year"
+
+[OutputVariables]
+WindSpeed= "WindSpeed"
+
+
+[VarGrid]
+GridProj= "+proj=longlat"
```
Comment (dpaolella) [inline inmap/configExampleGEOSChem.toml]: Why is this projection different than the one in the WRFChem config file?

Comment (dpaolella) [inline inmap/configExampleGEOSChem.toml]: Why is this projection different than the one in the WRFChem config file?

File: preproc.go [hunk 2]
```diff
@@ -940,6 +952,15 @@ func stabilityMixingChemistry(LayerHeights *sparse.DenseArray, pblhFunc, ustarFu
 	}
 }
 
+func temperatureToTheta(T, p float64) float64 {
+	const (
+		po    = 101300. // Pa, reference pressure
```
Comment (dpaolella) [inline preproc.go]: I'm guessing there's a reason that this is 101300 Pa and not 101325 Pa = 1 atm.

Comment (ctessum) [inline preproc.go]: Probably. I wasn't the one that wrote it originally.

File: inmap/configExampleGEOSChem.toml (hunk 1 above)
Comment (ctessum) [inline inmap/configExampleGEOSChem.toml]: GEOS-Chem uses a lat-lon grid, as I understand.

**Comment by ctessum** [comment]
Thanks!

**Pull request merged**